app/controllers/api/functions.php

```diff
@@ -1290,3 +1296,231 @@
+    ->desc('Get Variable')
+    ->groups(['api', 'functions'])
+    ->label('scope', 'functions.read')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'functions')
+    ->label('sdk.method', 'getVariable')
+    ->label('sdk.description', '/docs/references/functions/get-variable.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_VARIABLE)
+    ->param('functionId', null, new UID(), 'Function unique ID.', false)
+    ->param('variableId', null, new UID(), 'Variable unique ID.', false)
+    ->inject('response')
+    ->inject('dbForProject')
+    ->action(function (string $functionId, string $variableId, Response $response, Database $dbForProject) {
+        $function = $dbForProject->getDocument('functions', $functionId);
+
+        if ($function->isEmpty()) {
+            throw new Exception(Exception::FUNCTION_NOT_FOUND);
+        }
+
+        $variable = $dbForProject->findOne('variables', [
+            Query::equal('_uid', [$variableId]),
+            Query::equal('functionInternalId', [$function->getInternalId()]),
+        ]);
+
+        if ($variable === false || $variable->isEmpty()) {
+            throw new Exception(Exception::VARIABLE_NOT_FOUND);
+        }
+
+        $response->dynamic($variable, Response::MODEL_VARIABLE);
+    });
+
+App::put('/v1/functions/:functionId/variables/:variableId')
+    ->desc('Update Variable')
+    ->groups(['api', 'functions'])
+    ->label('scope', 'functions.write')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'functions')
+    ->label('sdk.method', 'updateVariable')
+    ->label('sdk.description', '/docs/references/functions/update-variable.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_VARIABLE)
+    ->param('functionId', null, new UID(), 'Function unique ID.', false)
+    ->param('variableId', null, new UID(), 'Variable unique ID.', false)
+    ->param('key', null, new Text(255), 'Variable key. Max length: 255 chars.', true)
+    ->param('value', null, new Text(8192), 'Variable value. Max length: 8192 chars.', true)
+    ->inject('response')
+    ->inject('dbForProject')
+    ->action(function (string $functionId, string $variableId, ?string $key, ?string $value, Response $response, Database $dbForProject) {
+        if (empty($key) && empty($value)) {
+            throw new Exception(Exception::VARIABLE_MISSING_PAYLOAD, 'Missing key or value. Define at least one.');
+        }
+
+        $function = $dbForProject->getDocument('functions', $functionId);
+
+        if ($function->isEmpty()) {
+            throw new Exception(Exception::FUNCTION_NOT_FOUND);
+        }
+
+        $variable = $dbForProject->findOne('variables', [
+            Query::equal('_uid', [$variableId]),
```

Comment:
we shouldn't use internal column names here, we should use `$id`

Reply:
Updated on multiple places.